[APO-2162] Update vellum json encoder to handle pydantic fieldinfo serialization

NgoHarrison opened the pull request.

Commit:
[APO-2162] Update vellum json encoder to handle pydantic fieldinfo serialization

vincent0426 commented on src/vellum/utils/tests/test_json_encoder.py:
```diff
@@ -0,0 +1,70 @@
+        # Note: When a FieldInfo has default_factory instead of default,
+        # the default attribute will be PydanticUndefined, which should serialize to None
+        # The actual default_factory is only called when instantiating a model
+        assert result == '{"test": null}'
```
wouldn't this make pushing from sdk fails to serialize into default_factory and loss the value

devin-ai-integration[bot] committed:
Handle Pydantic v1 Undefined sentinel in FieldInfo serialization

- Add support for pydantic.fields.Undefined (v1) in addition to PydanticUndefined (v2)
- Update FieldInfo serialization logic to check for both v1 and v2 undefined sentinels
- Improve test comment to clarify default_factory serialization behavior

Addresses PR feedback from chatgpt-codex-connector and vincent0426

Co-Authored-By: harrison@vellum.ai <harrison.ngo719@gmail.com>

devin-ai-integration[bot] committed:
Add test demonstrating BaseModel with default_factory serializes correctly

This test shows that when pushing workflows to Vellum, default_factory values
are preserved because Pydantic invokes the factory during model instantiation.
The VellumJsonEncoder serializes the resulting model instance (with computed values),
not the FieldInfo metadata objects.

Co-Authored-By: harrison@vellum.ai <harrison.ngo719@gmail.com>
